### PR TITLE
Fix Carthage

### PR DESCRIPTION
**Pull request by jamesrb1**
It deletes all non-RevenueCat schemes which I assume are not needed by Carthage, so this should save a bit of time over what it was doing before too because it won't spend time on them.

https://app.circleci.com/pipelines/github/RevenueCat/purchases-ios/19899/workflows/a895e06e-305d-439d-9f99-3047a0375594

image

### CODE REVIEW

File: Tests/InstallationTests/CarthageInstallation/fastlane/Fastfile [hunk 1]
```diff
@@ -16,6 +16,14 @@ lane :installation_tests do
     # doesn't try to build the other installation tests
     sh "carthage", "update", "--no-build"
     sh "rm", "-rf", "Carthage/Checkouts/purchases-root/Tests/InstallationTests/"
+
+    # Carthage builds all schemes including ones we don't need, so let's nuke them before proceeding
+    schemes_directory = "Carthage/Checkouts/purchases-root/RevenueCat.xcodeproj/xcshareddata/xcschemes"
+    Dir.glob("#{schemes_directory}/*.xcscheme").each do |file_path|
+      next if File.basename(file_path) == "RevenueCat.xcscheme"
```
Comment (aboedo) [inline Tests/InstallationTests/CarthageInstallation/fastlane/Fastfile:23]: does this mean that we're skipping the test for RevenueCatUI?

**Review by aboedo — APPROVED** [review]
Looks good, but maybe we can allow-list revenueCatUI too?

File: Tests/InstallationTests/CarthageInstallation/fastlane/Fastfile (hunk 1 above)
Comment (jamesrb1) [reply Tests/InstallationTests/CarthageInstallation/fastlane/Fastfile:23]: RevenueCatUI scheme is only defined in the Package file, which is ignored by Carthage.

Comment (jamesrb1) [reply Tests/InstallationTests/CarthageInstallation/fastlane/Fastfile:23]: Oh wait that might be wrong, it's there but not shown? Anyway, this is the directory it searches, so it shouldn't affect RevenueCatUI.
```
jamesborthwick@Jamess-MacBook-Pro-2 xcschemes % ls -l 
total 48
-rw-r--r--@ 1 jamesborthwick  staff  4978 13 Apr 14:02 BackendIntegrationTests.xcscheme
-rw-r--r--@ 1 jamesborthwick  staff  3591 13 Apr 14:02 ReceiptParser.xcscheme
-rw-r--r--@ 1 jamesborthwick  staff  7220 14 Jun 14:47 RevenueCat.xcscheme
-rw-r--r--@ 1 jamesborthwick  staff  2609 20 Jun 10:21 RevenueCatUIDev.xcscheme
```

**Pull request merged**